lib/mix/lib/mix/tasks/test.ex

```diff
@@ -175,6 +175,24 @@ defmodule Mix.Tasks.Test do
   def run(args) do
     {opts, files} = OptionParser.parse!(args, strict: @switches)
 
+    if opts[:listen_on_stdin] do
+      try do
+        Process.flag(:trap_exit, true)
```

Comment:
Is trap_exit required? I am afraid it will hide actual exceptions. If we need to do like this, it would maybe be preferable to do it all on another process that always exits and use monitor to see the actual error.